packages/opentelemetry-basic-tracer/src/export/BatchSampledSpanProcessor.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Logger, TraceOptions } from '@opentelemetry/types';
+import { SpanProcessor } from './SpanProcessor';
+import { SpanExporter } from './SpanExporter';
+import { Span } from '../Span';
+import { ReadableSpan } from './ReadableSpan';
+import { NoopLogger } from '@opentelemetry/core';
+import { BufferConfig } from '../types';
+
+const DEFAULT_BUFFER_SIZE = 100;
+const DEFAULT_BUFFER_TIMEOUT = 20000;
+
+/**
+ * Implementation of the {@link SpanProcessor} that batches spans exported by
+ * the SDK then pushes them to the exporter pipeline.
+ */
+export class BatchSampledSpanProcessor implements SpanProcessor {
+  private readonly _logger: Logger;
+  private readonly _bufferSize: number;
+  private readonly _bufferTimeout: number;
+  private _resetTimeout = false;
+  private _bufferTimeoutInProgress = false;
+  private _spansDataList: ReadableSpan[] = [];
+
+  constructor(private readonly exporter: SpanExporter, config: BufferConfig) {
```

Comment:
I'm slightly confused here about the tracer-processor-exporter relationship. If the tracer has one processor and the processor has ONE exporter, how you can have multiple exporters? Shouldn't the processor work with multiple exporters?

Reply:
I'm also confused about exactly what a processor is supposed to be in this case and the relationship with the exporter. I would think they are unrelated concepts, and vendors should be able to add any number of processors and/or exporters independently.

Reply:
> If the tracer has one processor and the processor has ONE exporter, how you can have multiple exporters? Shouldn't the processor work with multiple exporters?

Makes sense to me, I have changed the `SpanProcessor` to work with multiple exporters. 

> vendors should be able to add any number of processors and/or exporters independently.

I have exposed `registerExporter()` method on tracer to add any number of exporters and for now, `SpanProcessor` is default to `BatchSampledSpanProcessor`. Maybe later will allow vendors/users to configure `SpanProcessor` also. WDYT?

Reply:
Why is the processor even needed? What is the purpose of the processor?